cockroachcloud/egress-perimeter-controls.md

```diff
@@ -0,0 +1,327 @@
+
+    {% include_cached copy-clipboard.html %}
+    ~~~yaml
+    ---
+    name: "roach-buckets"
+    type: "FQDN"
+    destination: "storage.googleapis.com"
+    ports: [443, 80]
+    paths:
+        - "/customer-manged-bucket-1/*" # replace with bucket path
+        - "/customer-manged-bucket-2/*" # replace with bucket path
+    description: 'egress for GCP storage buckets'
+    ~~~
+
+1. Next, create a YAML manifest for a CIDR-based rule.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~yaml
+    --- # egress-rule2.yml
+    name: "roach-kafka"
+    type: "CIDR"
+    destination: "123.34.62.123/32" # replace with Kafka cluster CIDR range
+    ports: [443]
+    paths: []
+    description: 'egress for Kafka'    
+    ~~~
+
+1. Use Ruby (or another technique), to compile human-editable YAML into API-parsable JSON:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    ruby -ryaml -rjson -e 'puts(YAML.load(ARGF.read).to_json)' < egress-rule1.yml > egress-rule1.json
+    ruby -ryaml -rjson -e 'puts(YAML.load(ARGF.read).to_json)' < egress-rule2.yml > egress-rule2.json
+    ~~~
+
+1. Use the shell utility `jq` to inspect the JSON payload:
+    
+    {{site.data.alerts.callout_info}}
+    On macOS, you can install `jq` from Homebrew: `brew install jq`
+    {{site.data.alerts.end}}
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    cat egress-rule1.json |jq
+    cat egress-rule2.json |jq
+    ~~~
+
+    ~~~json
+    {
+      "name": "roach-buckets",
+      "type": "FQDN",
+      "destination": "storage.googleapis.com",
+      "ports": [
+        443,
+        80
+      ],
+      "paths": [
+        "/customer-manged-bucket-1/*",
+        "/customer-manged-bucket-2/*"
+      ],
+      "description": "egress for GCP storage buckets"
+    }
+    {
+      "name": "roach-kafka",
+      "type": "CIDR",
+      "destination": "123.34.62.123/32",
+      "ports": [
+        443
+      ],
+      "paths": [],
+      "description": "egress for Kafka"
+    }
+    ~~~
+
+1. Make a `POST` request to create each rule from the JSON payload.
+    
+    [API spec]
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    curl --request POST \
+    --header "Authorization: Bearer $CC_API_KEY" \
+    --url "https://management-staging.crdb.io/api/v1/clusters/$CLUSTER_ID/networking/egress-rules" \
+    --data "@egress-rule1"
+
+    curl --request POST \
+    --header "Authorization: Bearer $CC_API_KEY" \
+    --url "https://management-staging.crdb.io/api/v1/clusters/$CLUSTER_ID/networking/egress-rules" \
+    --data "@egress-rule2"
+    ~~~
+
+    ~~~json
+
+    ~~~
+
+    {{site.data.alerts.callout_danger}}
+    Your cluster's firewall behavior is not updated instantly when you submit the API request. After submitting the request, [check your egress rules](#check-a-clusters-egress-rules-allowed-destinations) to confirm that the new rules have been created.
```

Comment:
I'd reword this to `s/Your cluster's firewall behavior is not updated instantly when you submit the API request/Your cluster's firewall behavior is enforced asynchronously after the API response.`

Reply:
thx, much better!